distribution/src/resources/drill-config.sh

```diff
@@ -359,6 +366,7 @@ if [ -z "$JAVA_HOME" ]; then
       [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE"
     done
     JAVA_HOME="$( cd -P "$( dirname "$SOURCE" )" && cd .. && pwd )"
+    JAVA = $SOURCE/$JAVA_BIN
```

Comment:
$SOURCE already includes bin/java, so concatenating with $JAVA_BIN will give you:
`/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.111-1.b15.el7_2.x86_64/jre/bin/java/java`.
I guess you should concatenate $JAVA_BIN to $SOURCE dirname.